src/formatDate.js

```diff
@@ -8,7 +8,56 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+
+  const dateParts = date.split(separatorFrom);
+
+  let year, month, day;
+
+  fromFormat.forEach((part, index) => {
+    if (part === 'YYYY' || part === 'YY') {
+      year = dateParts[index];
+    } else if (part === 'MM') {
+      month = dateParts[index];
+    } else if (part === 'DD') {
+      day = dateParts[index];
+    }
+  });
+
+  const convertYear = (inputYear, targetFormat) => {
+    if (targetFormat === 'YYYY') {
+      if (inputYear.length === 2) {
+        return Number(inputYear) < 30 ? '20' + inputYear : '19' + inputYear;
+      }
+
+      return inputYear;
+    } else if (targetFormat === 'YY') {
+      return inputYear.length === 4 ? inputYear.slice(-2) : inputYear;
+    }
+  };
```

Comment:
The `convertYear` function is declared inside the `formatDate` function, which means a new instance of `convertYear` will be created each time `formatDate` is called. For better performance, consider moving `convertYear` outside of `formatDate` so it's only created once.